src/score.js

```diff
@@ -47,31 +30,48 @@ const sayCalcFailed = (isAskingForOtherUser = false) => `Sorry, an error occurre
 
 /**
  * @summary calculates the score
- * @param {User} user API user object
- * @param {Badge[]} badges user badges
+ * @param {ApiUser} user API user object
+ * @param {Badge[]} userBadges user badges
  * @param {ElectionBadge[]} electionBadges election badges
+ * @param {Election} election current election
+ * @param {boolean} [isSO] is Stack Overflow election
  * @returns {{
  *  score: number,
  *  missing: {
- *      badges: ElectionBadge[]
- *  }
+ *      badges: {
+ *          election: ElectionBadge[],
+ *          required: ElectionBadge[]
+ *      }
+ *  },
+ *  isEligible: boolean
  * }}
  */
-export const calculateScore = (user, badges, electionBadges) => {
+export const calculateScore = (user, userBadges, electionBadges, election, isSO = false) => {
```

Comment:
This significantly reduces the number of times we violate SRP when calculating the score :)